src/backend/columnar/cstore_tableam.c

```diff
@@ -1286,6 +1290,135 @@ columnar_handler(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * CitusCreateAlterColumnarTableSet generates a portable
+ */
+static char *
+CitusCreateAlterColumnarTableSet(char *qualifiedRelationName,
+								 const ColumnarOptions *options)
+{
+	StringInfoData buf = { 0 };
+	initStringInfo(&buf);
+
+	appendStringInfo(&buf,
+					 "SELECT alter_columnar_table_set(%s, "
+					 "block_row_count => %d, "
+					 "stripe_row_count => %lu, "
+					 "compression => %s);",
+					 quote_literal_cstr(qualifiedRelationName),
+					 options->blockRowCount,
+					 options->stripeRowCount,
+					 quote_literal_cstr(CompressionTypeStr(options->compressionType)));
+
+	return buf.data;
+}
+
+
+/*
+ * ColumnarTableDDLContext holds the instance variable for the TableDDLCommandFunction
+ * instance described below.
+ */
+typedef struct ColumnarTableDDLContext
+{
+	char *schemaName;
+	char *relationName;
+	ColumnarOptions options;
+} ColumnarTableDDLContext;
+
+
+/*
+ * GetTableDDLCommandColumnar is an internal function used to turn a
+ * ColumnarTableDDLContext stored on the context of a TableDDLCommandFunction into a sql
+ * command that will be executed against a table. The resulting command will set the
+ * options of the table to the same options as the relation on the coordinator.
+ */
+static char *
+GetTableDDLCommandColumnar(void *context)
+{
+	ColumnarTableDDLContext *tableDDLContext = (ColumnarTableDDLContext *) context;
+
+	char *qualifiedShardName = quote_qualified_identifier(tableDDLContext->schemaName,
+														  tableDDLContext->relationName);
+
+	return CitusCreateAlterColumnarTableSet(qualifiedShardName,
+											&tableDDLContext->options);
+}
+
+
+/*
+ * GetShardedTableDDLCommandColumnar is an internal function used to turn a
+ * ColumnarTableDDLContext stored on the context of a TableDDLCommandFunction into a sql
+ * command that will be executed against a shard. The resulting command will set the
+ * options of the shard to the same options as the relation the shard is based on.
+ */
+static char *
+GetShardedTableDDLCommandColumnar(uint64 shardId, char *schemaName, void *context)
```

Comment:
do we need schemaName in `TableDDLShardedFunction `? We don't seem to use it.

Reply:
This function needs to satisfy the signature for the function pointers in `TableDDLCommand`. The `schemaName` is there for legacy reasons and ease of refactor, atleast for the `TableDDLCommandString` variant.

We could change the function pointers here to not pass through the `schemaName` and work towards a future where the schemaName is refactored out of the current commands and `worker_apply_shard_ddl_command`.

Just removing it here will fail compilation due to incompatible function pointers.

What are the chances we would need the `schemaName` in a future implementation? I guess pretty low right? In that case I think it might indeed be better to remove it from the signature already.

ps: we will still need to pass it in for the `DDLJob` transformation as it would still be required from the public interface of the `TableDDLCommand`.

Reply:
have deprecated the `schemaName`. Only allowed for backwards compatibility reasons in `TableDDLCommandString`. Added a comment to express on the dispatcher `GetShardedTableDDLCommand` where it is still present.